orttraining/orttraining/python/training/execution_agent.py

```diff
@@ -77,9 +77,17 @@ class TrainingAgent(object):
     This is the main class used to run an ORTModule model training.
     """
 
-    def __init__(self, path_or_bytes, session_options=None, providers=None, provider_options=None):
+    def __init__(self, path_or_bytes, fw_feed_names, fw_fetches_names, fw_outputs_device_info,
```

Comment:
`fw_feed_names, fw_fetches_names, fw_outputs_device_info,bw_feed_names, bw_fetches_names, bw_outputs_device_info` are all tightly related. How about merging them on a `TrainingAgentConfig` class or something simuilar?

Having a constructor with 10 parameters is usually a sign for refactoring

Reply:
This is an internal API with one callsite, so there's very little benefit (and non-zero runtime cost!) in creating another wrapper class to pack arguments together. Please keep is as-is.

Reply:
thanks @mrry